### PR TITLE
A0-2418: fix scores regression

### PR DESCRIPTION
**Pull request by fbielejec**
# Description

# Checklist:

- I have added tests

### CODE REVIEW

File: contracts/env/dev [hunk 1]
```diff
@@ -5,13 +5,19 @@ export AUTHORITY=5GrwvaEF5zXb26Fz9rcQpDWS57CtERHpNehXCPcNoHGKutQY
 export AUTHORITY_SEED=//Alice
 
 # how long do the games live for (in blocks)
-export LIFETIME=20
+export LIFETIME=900
```
Comment (obrok) [inline contracts/env/dev:8]: This lifetime cannot be so long for e2e tests - we're waiting for the button to die multiple times, they would take forever. If you want to keep this file for dev, then we need another one with a short lifetime for tests.

**Review by obrok — CHANGES_REQUESTED** [review]
The scores and payouts still have a different scale, so an update to the e2e tests is still neded:

```
525f2b6fc3946a8708 (5GRRztGq...), name: Some("ButtonPressed"), data: {"when": UInt(250), "score": UInt(4), "by": Literal("5G7LFy8yqvPmNVmDKYU44W4WKpcBH4esbyZzYJdWYJwrHwbA")} })
Assertion failed at src/test/button_game/mod.rs:475:5:
  assert!( reward_token.balance_of(&conn, player.account_id()).await? == first_presser_score )
with expansion:
  20800000000000 == 4
```

**Review by obrok — CHANGES_REQUESTED** [review]
I'm still getting

```
[2023-04-24T17:06:43Z INFO  aleph_e2e_client::test::button_game::helpers] Received contract event Ok(ContractEvent { contract: 1a56e4fa8ccd25b2007a444cf8b8f31e52092768bccaed8418b52bf3ccff3f7d (5CfEv31F...), name: Some("RewardMinted"), data: {"when": UInt(507), "reward_token": Literal("5CU12oyGxPtfVBhzc8g6CuxDSqstayUsdwwr7LPcbjQ4GcnH"), "to": Literal("5CaEj5GkTupe4BYg58rPDGekVYJ2FiWNAmNUruvnjsaze7N7"), "amount": UInt(0)} })
Assertion failed at src/test/button_game/mod.rs:513:5:
  assert!( reward_token.balance_of(&conn, player.account_id()).await? == total_score + pressiah_score )
with expansion:
  3000000000000 == 3750000000000
  ```
  
  so it seems like something is not right with the pressiah reward. Note the `RewardMinted` event in the first line has `"amount": UInt(0)`.

File: contracts/env/dev (hunk 1 above)
Comment (fbielejec) [reply contracts/env/dev]: ok, scratched

**Comment by fbielejec** [comment]
> The scores and payouts still have a different scale, so an update to the e2e tests is still neded:
> 
> ```
> 525f2b6fc3946a8708 (5GRRztGq...), name: Some("ButtonPressed"), data: {"when": UInt(250), "score": UInt(4), "by": Literal("5G7LFy8yqvPmNVmDKYU44W4WKpcBH4esbyZzYJdWYJwrHwbA")} })
> Assertion failed at src/test/button_game/mod.rs:475:5:
>   assert!( reward_token.balance_of(&conn, player.account_id()).await? == first_presser_score )
> with expansion:
>   20800000000000 == 4
> ```

they are passing for me now. I take no resp for the flakiness of the CI

**Comment by fbielejec** [comment]
> I'm still getting
> 
> ```
> [2023-04-24T17:06:43Z INFO  aleph_e2e_client::test::button_game::helpers] Received contract event Ok(ContractEvent { contract: 1a56e4fa8ccd25b2007a444cf8b8f31e52092768bccaed8418b52bf3ccff3f7d (5CfEv31F...), name: Some("RewardMinted"), data: {"when": UInt(507), "reward_token": Literal("5CU12oyGxPtfVBhzc8g6CuxDSqstayUsdwwr7LPcbjQ4GcnH"), "to": Literal("5CaEj5GkTupe4BYg58rPDGekVYJ2FiWNAmNUruvnjsaze7N7"), "amount": UInt(0)} })
> Assertion failed at src/test/button_game/mod.rs:513:5:
>   assert!( reward_token.balance_of(&conn, player.account_id()).await? == total_score + pressiah_score )
> with expansion:
>   3000000000000 == 3750000000000
> ```
> 
> so it seems like something is not right with the pressiah reward. Note the `RewardMinted` event in the first line has `"amount": UInt(0)`.

```
running 7 tests
test test::button_game::marketplace ... ok
test test::button_game::button_game_reset ... ok
test test::button_game::back_to_the_future has been running for over 60 seconds
test test::button_game::early_bird_special has been running for over 60 seconds
test test::button_game::simple_dex has been running for over 60 seconds
test test::button_game::the_pressiah_cometh has been running for over 60 seconds
test test::button_game::wrapped_azero has been running for over 60 seconds
test test::button_game::early_bird_special ... ok
test test::button_game::the_pressiah_cometh ... ok
test test::button_game::back_to_the_future ... ok
test test::button_game::wrapped_azero ... ok
test test::button_game::simple_dex ... ok

test result: ok. 7 passed; 0 failed; 0 ignored; 0 measured; 37 filtered out; finished in 247.09s
```

Did you pull? Seems like there was some delay

File: e2e-tests/src/test/button_game/mod.rs [hunk 2]
```diff
@@ -487,28 +487,30 @@ async fn button_game_play<F: Fn(u128, u128, u128, u128)>(
...
+    let player_rewards = first_reward_amount + second_reward_amount;
+    assert!(reward_token.balance_of(&conn, player.account_id()).await? == player_rewards);
 
     wait_for_death(&conn, &button).await?;
     button.reset(&sign(&conn, &authority)).await?;
-    assert_recv_id(&mut events, "Reset").await;
+    let event = assert_recv_id(&mut events, "RewardMinted").await;
+    let_assert!(Some(&Value::UInt(pressiah_reward)) = event.data.get("amount"));
 
-    let pressiah_score = total_score / 4;
```
Comment (obrok) [inline e2e-tests/src/test/button_game/mod.rs:509]: This is a somewhat different test than before - now we depend on the events to tell us what's the reward of the pressiah, before we asserted that it's some specific number. Are we OK with that?

**Review by obrok — APPROVED** [review]
Yeah, my run/review was between the two commits. Looks good, left one question.

File: e2e-tests/src/test/button_game/mod.rs (hunk 2 above)
Comment (fbielejec) [reply e2e-tests/src/test/button_game/mod.rs:509]: It's just that the mapping of scores to the domain of rewards does not preserve that fact. 
(The score is a quarter of the totals but the reward (after mapping to a fixed domain) is not). 

tl;dr: given it would be a big overhaul to test that, and all we get is a test of dividing by 4, imo not worth the effort
